nautobot_golden_config/models.py

```diff
@@ -480,11 +477,12 @@ class GoldenConfigSetting(PrimaryModel):
         null=True,
         help_text="API filter in JSON format matching the list of devices for the scope of devices to be considered.",
     )
-    sot_agg_query = models.TextField(
-        null=False,
+    sot_agg_query = models.ForeignKey(
+        to="extras.GraphQLQuery",
+        on_delete=models.SET_NULL,
```

Comment:
@jeffkala @mzbroch what do you think, is this your expectation?

Reply:
I'd say it should "PROTECT".
Deleting GQL entries could have an implicit effect impacting GC, I'd be for a warning message.

Reply:
ok, will fix in a subsequent PR